plugin/src/test/java/org/opensearch/ml/tools/ToolIntegrationWithLLMTest.java

```diff
@@ -63,9 +71,41 @@ public void stopMockLLM() {
     @After
     public void deleteModel() throws IOException {
         undeployModel(modelId);
+        waitModelUndeployed(modelId);
         deleteModel(client(), modelId, null);
     }
 
+    @SneakyThrows
+    private void waitModelUndeployed(String modelId) {
+        Predicate<Response> condition = response -> {
+            try {
+                Map<String, Object> responseInMap = parseResponseToMap(response);
+                String state = responseInMap.get(MLModel.MODEL_STATE_FIELD).toString();
+                return !state.equals(MLModelState.DEPLOYED.toString())
+                    && !state.equals(MLModelState.DEPLOYING.toString())
+                    && !state.equals(MLModelState.PARTIALLY_DEPLOYED.toString());
```

Comment:
Is undeployed state either MLModelState.UNDEPLOYED or MLModelState.DEPLOY_FAILED?

Reply:
https://github.com/opensearch-project/neural-search/blob/7c54c86e256cfb43d0e7623bda1f7590c4a18b0e/src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java#L932C1-L934C67

Reply:
I refer the logic from here https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java#L326

>Is undeployed state either MLModelState.UNDEPLOYED or MLModelState.DEPLOY_FAILED?

Yes, i think most of case should be these two. There have two more `REGISTERING`,`REGISTERED` allow to delete.

Reply:
Since we are undeploying the model, why we will get REGISTERING,REGISTERED state?

Reply:
Also from `isModelNotDeployed` function, the model state cannot be LOADED, LOADING and PARTIALLY_LOADED

Reply:
As we do undeploy first, the state could only be `UNDEPLOYED` and `DEPLOY_FAILED`, i have changed to check these two state.